compiler/rustc_target/src/spec/apple_base.rs

```diff
@@ -114,3 +114,8 @@ pub fn ios_sim_llvm_target(arch: &str) -> String {
     let (major, minor) = ios_deployment_target();
     format!("{}-apple-ios{}.{}.0-simulator", arch, major, minor)
 }
+
+pub fn macabi_llvm_target(arch: &str) -> String {
+    let (major, minor) = ios_deployment_target();
```

Comment:
I'm not sure whether it is a good way. But as an iOS developer, I believe it looks good. Because in Xcode, Mac Catalyst minimal version is highly related to minimal iOS version. For example, if I choose iOS 14.0, I can only pick macOS 11+ for this situation:

<img width="640" alt="image" src="https://user-images.githubusercontent.com/2396817/166129562-9683a885-11ca-43b2-aa20-e9054e2a2c88.png">

<img width="636" alt="image" src="https://user-images.githubusercontent.com/2396817/166129584-75cced66-a890-460a-a84f-a4a26d2ed47e.png">

(Copied from https://github.com/alexcrichton/cc-rs/pull/678)

Reply:
This looks like a good solution!